internal/arduino/libraries/librariesmanager/install.go

```diff
@@ -249,25 +249,44 @@ func (lmi *Installer) InstallGitLib(gitURL string, overwrite bool) error {
 	return nil
 }
 
-// parseGitURL tries to recover a library name from a git URL.
+// parseGitArgURL tries to recover a library name from a git URL.
 // Returns an error in case the URL is not a valid git URL.
-func parseGitURL(gitURL string) (string, plumbing.Revision, error) {
-	var res string
-	var rev plumbing.Revision
-	if strings.HasPrefix(gitURL, "git@") {
+func parseGitArgURL(argURL string) (string, string, plumbing.Revision, error) {
+	// On Windows handle paths with backslashes in the form C:\Path\to\library
+	if path := paths.New(argURL); path != nil && path.Exist() {
+		return path.Base(), argURL, "", nil
+	}
+
+	// Handle github-specific address in the form "git@github.com:arduino-libraries/SigFox.git"
+	if strings.HasPrefix(argURL, "git@github.com:") {
 		// We can't parse these as URLs
-		i := strings.LastIndex(gitURL, "/")
-		res = strings.TrimSuffix(gitURL[i+1:], ".git")
-	} else if path := paths.New(gitURL); path != nil && path.Exist() {
-		res = path.Base()
-	} else if parsed, err := url.Parse(gitURL); parsed.String() != "" && err == nil {
-		i := strings.LastIndex(parsed.Path, "/")
-		res = strings.TrimSuffix(parsed.Path[i+1:], ".git")
-		rev = plumbing.Revision(parsed.Fragment)
-	} else {
-		return "", "", errors.New(i18n.Tr("invalid git url"))
-	}
-	return res, rev, nil
+		argURL = "https://github.com/" + strings.TrimPrefix(argURL, "git@github.com:")
+	}
```

Comment:
Any reason why we are restricting only to `github` provider?
Others allow us to do the same thing like:
- GitLab (https://gitlab.com/painlessMesh/painlessMesh)
- bitbucket (https://bitbucket.org/jezhill/signalacquisition/src/main/)

Looking at the git docs under `The SSH Protocol` section, it should be a git standard.
https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols.

Reply:
I've added GitLab and bitbucket too.

By looking at the documentation:

> The SSH Protocol
> A common transport protocol for Git when self-hosting is over SSH. This is because SSH access to servers is already set up in most places — and if it isn’t, it’s easy to do. SSH is also an authenticated network protocol and, because it’s ubiquitous, it’s generally easy to set up and use.
>
> To clone a Git repository over SSH, you can specify an ssh:// URL like this:
>
> ```
> $ git clone ssh://[user@]server/project.git
> ```
> Or you can use the shorter scp-like syntax for the SSH protocol:
>
> ```
> $ git clone [user@]server:project.git
> ```
> In both cases above, if you don’t specify the optional username, Git assumes the user you’re currently logged in as.

It does say that we can use SSH but it does not mandate that the username must be `git@hostname` and neither that the HTTPS URL should match the address, I think we cannot generalize these exceptions.